pkg/autoscaler/k8sclient/k8sclient_test.go

```diff
@@ -76,6 +76,76 @@ func TestGetScaleTarget(t *testing.T) {
 	}
 }
 
+func TestGetScaleTarget(t *testing.T) {
+	testCases := []struct {
+		target          string
+		expScaleTargets *scaleTargets
+		expError        bool
+	}{
+		{
+			"deployment/anything",
+			&scaleTargets{
+				targets: []target{
+					{kind: "deployment", name: "anything"},
+				},
+			},
+			false,
+		},
+		{
+			"deployment/first,deployment/second",
+			&scaleTargets{
+				targets: []target{
+					{kind: "deployment", name: "first"},
+					{kind: "deployment", name: "second"},
+				},
+			},
+			false,
+		},
+		{
+			"deployment/first, deployment/second",
+			&scaleTargets{
+				targets: []target{
+					{kind: "deployment", name: "first"},
+					{kind: "deployment", name: "second"},
+				},
+			},
+			false,
+		},
+		{
+			"deployment/first deployment/second",
+			&scaleTargets{
+				targets: []target{
+					{kind: "deployment", name: "first"},
+					{kind: "deployment", name: "second"},
+				},
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		res, err := getScaleTargets(tc.target, "default")
+		if err != nil && !tc.expError {
+			t.Errorf("Expect no error, got error for target: %v", tc.target)
+			continue
+		} else if err == nil && tc.expError {
+			t.Errorf("Expect error, got no error for target: %v", tc.target)
+			continue
+		}
+		if len(res.targets) != len(tc.expScaleTargets.targets) && !tc.expError {
+			t.Errorf("Expected targets vs resulted targets should be the same length: %v vs %v", len(tc.expScaleTargets.targets), len(res.targets))
+			continue
+		}
+		for idx := 1; idx <= len(res.targets); idx++ {
+			if res.targets[0].kind != tc.expScaleTargets.targets[0].kind ||
```

Comment:
Did you mean to put `[idx]` here? Or we are always verifying the first element.

As a golang nit, `range` is what we typically use to iterate through a slice.
```golang
for i, target := range res.targets {
   ...
}
```

Reply:
Yes that's true. That came from the upstream branch that the changes took place.